File: content/showcase/8-power-reflection/any-reflection.lacking.ts
// a function is an object: its name, arity and source
// text can be read back; its types are erased
function power(b: number, n: number): number {
  return b ** n;
}

console.log(power.name, power.length); // power 2
console.log(power.toString());  // the source text

File: content/showcase/10-power-reflection/any-reflection.lacking.py
# lacking: source comes back as text and bytecode, not nodes; an expression runs before it can be read
# a function is an object: its signature, its source
# and its bytecode can all be read back
import inspect, dis

def power(b: int, n: int) -> int:
    return b ** n

print(inspect.signature(power))  # the signature
print(inspect.getsource(power))  # the source text
dis.dis(power)                   # the bytecode
